### ISSUE TITLE
Ignore brute force warnings safely

### ISSUE THREAD

**Issue by kendyblack** [wazuh/wazuh#10928]
How to bypass an attack that is considered safe. I received an insider brute force warning, but have determined it to be a non-hazardous warning. How can I whitelist it for wazuh to not receive warnings anymore?

**Comment by jctello** [comment]
Hi @kendyblack you may create a child rule to silence a subset of events that match this rule and also fulfill the criteria you used to determined it was non-hazardous.

For example, if you're seeing rule `5712: sshd: brute force trying to get access to the system.`  being triggered from a specific IP you may add the following to your custom rules:

```
  <rule id="100002" level="0">
    <if_sid>5712</if_sid>
    <srcip>8.8.8.8</srcip>
    <description>Brute force attempt from non-hazardous IP 8.8.8.8.</description>
    <group>authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,</group>
  </rule>
```

If you wish to fully disable this rule for all events you may also overwrite the original rule by adding this to your custom rules:
```
  <rule id="5712" level="0" frequency="8" timeframe="120" ignore="60" overwrite="yes">
    <if_matched_sid>5710</if_matched_sid>
    <description>Silenced sshd brute-force rule</description>
    <same_source_ip />
  </rule>
```
Note that I have changed the level to 0, added the `overwrite="yes"` parameter, modified the description and removed mapping information.

You may find documentation on how to modify the rules here: https://documentation.wazuh.com/current/user-manual/ruleset/custom.html

You can also edit the ruleset using the Wazuh Kibana plugin by selecting Management from the Wazuh menu and then Rules:
![image](https://user-images.githubusercontent.com/36071202/142995508-054c4ed4-9a6a-4641-b2b3-bf18d54ebcb0.png)

and once there clicking on the **Custom rules** button and selecting the `local_rules.xml` file:
![image](https://user-images.githubusercontent.com/36071202/142995642-cd8b992e-e9b4-4a7c-9e13-c1f035ab3d60.png)

Be sure to place the custom rules within a `<group>` tag.

**Comment by kendyblack** [comment]
![image](https://user-images.githubusercontent.com/66987635/142997180-b3a3fa03-c10b-4a88-bab0-66f046beebda.png)
My problem is as shown in the picture. Is it possible to whitelist by url? Because of the internal call, I can't see the IP

**Comment by jctello** [comment]
Yes, for this specific event for example you can either overwrite the original rule to remove `/server-status` as one of the suspicious URLs:

```
  <rule id="31516" level="6" overwrite="yes">
    <if_sid>31100</if_sid>
    <url>.swp$|.bak$|/.htaccess|/.ssh|/.history|/wallet.dat</url>
    <description>Suspicious URL access.</description>
    <mitre>
      <id>T1055</id>
    </mitre>
   <group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
   </rule>
```
or create a specific child rule for `/server-status?auto` so this particular url does not trigger rule 31516:
```
  <rule id="100002" level="3">
    <if_sid>31516</if_sid>
    <url>^/server-status?auto</url>
    <description>Non-hazardous request to server-status</description>
    <mitre>
      <id>T1055</id>
    </mitre>
   <group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
```
In this case I'm intentionally logging the event as a level 3 event, but if you wish you may set it to 0.

**Comment by kendyblack** [comment]
![image](https://user-images.githubusercontent.com/66987635/143219417-0a570a46-97d5-4d13-80e6-e3396ca6edef.png)
By the way, I have a problem with showing the source of the attack when I receive an alert

**Comment by jctello** [comment]
Hi @kendyblack in this case the source log does not include IP information, if it would be the case it would appear in place of the first of three `-` characters. The common log format is described [here](https://httpd.apache.org/docs/2.4/logs.html#common). 

To address this it would be necessary to determine why your web server is not showing the source IP.
